File: Algorithms/Implementation/Almost Sorted.php
<?php

/*
 * Complete the 'almostSorted' function below.
 *
 * The function accepts INTEGER_ARRAY arr as parameter.
 */

function almostSorted($arr) {
    # Write your code here
    $n = count($arr);
    $sortedArr = $arr;
    sort($sortedArr);

    if ($arr === $sortedArr) {
        echo "yes\n";
        return;
    }

    $l = 0;
    $r = $n - 1;

    while ($l < $n && $arr[$l] === $sortedArr[$l]) {
        $l++;
    }
    while ($r >= 0 && $arr[$r] === $sortedArr[$r]) {
        $r--;
    }

    list($arr[$l], $arr[$r]) = array($arr[$r], $arr[$l]);

    if ($arr === $sortedArr) {
        echo "yes\n";
        echo "swap " . ($l + 1) . " " . ($r + 1) . "\n";
        return;
    }

    list($arr[$l], $arr[$r]) = array($arr[$r], $arr[$l]);

    $subArray = array_slice($arr, $l, $r - $l + 1);
    $reversedSubArray = array_reverse($subArray);

    if (array_merge(array_slice($arr, 0, $l), $reversedSubArray, array_slice($arr, $r + 1)) === $sortedArr) {
        echo "yes\n";
        echo "reverse " . ($l + 1) . " " . ($r + 1) . "\n";
        return;
    }

    echo "no\n";
}

$n = intval(trim(fgets(STDIN)));

$arr_temp = rtrim(fgets(STDIN));

$arr = array_map('intval', preg_split('/ /', $arr_temp, -1, PREG_SPLIT_NO_EMPTY));

almostSorted($arr);

File: Algorithms/Implementation/Almost Sorted.py
import os
import sys


#
# Complete the 'almostSorted' function below.
#
# The function accepts INTEGER_ARRAY arr as parameter.
#


def almostSorted(arr):
    # Write your code here
    n = len(arr)
    sorted_arr = sorted(arr)

    if arr == sorted_arr:
        print("yes")
        return

    l, r = 0, n - 1

    while l < n and arr[l] == sorted_arr[l]:
        l += 1

    while r >= 0 and arr[r] == sorted_arr[r]:
        r -= 1

    arr[l], arr[r] = arr[r], arr[l]

    if arr == sorted_arr:
        print("yes")
        print(f"swap {l + 1} {r + 1}")
        return

    arr[l], arr[r] = arr[r], arr[l]

    sub_array = arr[l : r + 1][::-1]
    if arr[:l] + sub_array + arr[r + 1 :] == sorted_arr:
        print("yes")
        print(f"reverse {l + 1} {r + 1}")
        return

    print("no")


if __name__ == "__main__":
    n = int(input().strip())

    arr = list(map(int, input().rstrip().split()))

    almostSorted(arr)
